lib/collection/request.js

```diff
@@ -251,6 +311,50 @@ _.assign(Request.prototype, /** @lends Request.prototype */ {
         this.url.removeQueryParams(params);
     },
 
+    /**
+     * Get the request size by computing the headers and body or using the
+     * actual content length header once the request is sent.
+     *
+     * @returns {Object}
+     */
+    size: function () {
+        var contentLength = this.headers.get(CONTENT_LENGTH),
+            requestTarget = this.url.getPathWithQuery(),
+            bodyString,
+            sizeInfo = {
+                body: 0,
+                header: 0,
+                total: 0
+            };
+
+        // if 'Content-Length' header is present, we take body as declared by
+        // the client(postman-request or user-defined). else we need to compute the same.
+        if (contentLength && util.isNumeric(contentLength)) {
+            sizeInfo.body = parseInt(contentLength, 10);
+        }
+        // otherwise, if body is defined, we calculate the length of the body
+        else if (this.body) {
+            // @note body.toString() returns E for formdata or file mode
```

Comment:
Need to attempt something for form data and file mode + append message in “source” property that computation is not accurate.

Reply:
Added to my ToDo. Also, we should consider setting our own form-data boundary to better control this?

Reply:
agreed regarding form data boundary.